trysyslog.c

```diff
@@ -2,8 +2,10 @@
 #include <sys/time.h>
 #include <syslog.h>
 
+int
```

Comment:
why `int` and `return 0` here?  Other try*.c functions (e.g., `tryflock.c`, `trymkffo.c`, `trysalen.c` don't do this, they use void.

Reply:
but they do _exit(). The return value must come from somewhere.

Reply:
We probably agree that at some point all our `main()`s should return `int`. I'm not understanding why it's better to make that change to these few `try*.c` ones. My thinking is, the problem this PR wants to solve is an internal inconsistency (specifically, that `main()` _almost_ always has _some_ return type). Upgrading to return `int` would make sense if it didn't also add internal consistency (specifically, with the other `try*.c` that already do have a return type). And I don't see any external forces (like the `try*` programs being executed and their exit codes being inspected) to change the balance here.

I appreciate that this is very small and in the right general direction, and that this discussion is far longer than the changes themselves. Still, unless I'm missing something, I think this part of the PR is the seed of another PR.